docs/rfcs/templates_solutions.md

```diff
@@ -0,0 +1,314 @@
+# Templates Solution Comparison
+
+The purpose of this document is to outline the primary use cases for
+submission templates and walk through potential solutions for the most common
+use case. Complexity, pros, and cons will be described for each solution.
+
+Hopefully this can help us decide how we want to implement data
+submission templates, MVP and beyond.
+
+## Use Cases
+
+1. The same template is needed by multiple studies for data submission
+2. A new template needs to be added to a study's existing templates to capture a new kind of data
+3. A template is deprecated and needs to be removed from a study so submitters stop using it
+4. An existing study template needs to be changed for that study (e.g. add col, remove col, rename col, change col type)
+
+## Most Common Use Case (4)
+
+1. CHOP admin creates initial templates:
+
+    ```
+    Participant Demographics (PDE)
+    Biospecimen Collection Manifest (BCM)
+    ```
+
+2. CHOP team iterates on development of [`PDE`, `BCM`] until they're happy with
+   first draft of templates
+
+3. CHOP admin assigns [`PDE`, `BCM`] to [`Study-CBTN`, `Study-A`] for use
+
+4. Data submitters begin using [`PDE`, `BCM`] to upload files to `Study-CBTN` and `Study-A`
+
+5. Changes to `BCM` (add col, remove col, rename a col) are needed for `Study-CBTN` but not `Study-A`
+
+## Solution 1: Templates
+
+### Model
+
+![Model](../_static/images/template-models-solution1.png)
+
+### Implementation
+
+1. Create Templates
+
+    ```
+    Templates
+    ---------
+    PDE, BCM
+    ```
+
+2. Add Templates to studies:
+
+    ```
+    Study-CBTN.templates = [PDE, BCM]
+    Study-A.templates = [PDE, BCM]
+    ```
+
+... People start using templates
+
+... Changes are required to BCM
+
+3. Copy `BCM` -> `BCM_V1`, Update `BCM_V1` with desired changes
+
+    ```
+    Templates
+    ---------
+    PDE, BCM, BCM_V1
+    ```
+
+4. Replace `BCM` with `BCM_V1` in CBTN
```

Comment:
what about all the documents in CBTN that are still using `BCM`?

Reply:
I haven't covered data migration when template definitions change. Still have to think about that